### PR TITLE
package gpt: make the EqualHeader return useful info

### PR DESCRIPTION
**Pull request by rminnich**
Instead of just 'things differ'

TODO: PartitionsDiffer

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

File: pkg/gpt/gpt.go [hunk 1]
```diff
@@ -74,20 +74,52 @@ func (g *GPT) String() string {
...
+	if p.Signature != b.Signature {
+		err = errAppend(err, "p.Signature(%#x) != b.Signature(%#x)", p.Signature, b.Signature)
+	}
+	if p.Revision != b.Revision {
+		err = errAppend(err, "p.Revision(%v) != b.Revision(%v)", p.Revision, b.Revision)
+	}
+	if p.HeaderSize != b.HeaderSize {
+		err = errAppend(err, "p.HeaderSize(%v) != b.HeaderSize(%v)", p.HeaderSize, b.HeaderSize)
+	}
+	if p.CurrentLBA != b.BackupLBA {
```
Comment (hugelgupf) [inline pkg/gpt/gpt.go]: Current != Backup?

Should this not be Current != Current?

File: pkg/gpt/gpt.go [hunk 2]
```diff
@@ -74,20 +74,52 @@ func (g *GPT) String() string {
...
+		err = errAppend(err, "p.CurrentLBA(%#x) != b.BackupLBA(%#x)", p.CurrentLBA, b.BackupLBA)
+	}
+	if p.BackupLBA != b.CurrentLBA {
```
Comment (hugelgupf) [inline pkg/gpt/gpt.go]: same?

File: pkg/gpt/gpt.go (hunk 1 above)
Comment (hugelgupf) [reply pkg/gpt/gpt.go]: I don't know anything about gpt, but this stands out.

Comment (rminnich) [reply pkg/gpt/gpt.go]: no. The one points to the other. It fooled me too at first.

File: pkg/gpt/gpt.go (hunk 2 above)
Comment (rminnich) [reply pkg/gpt/gpt.go]: Nope, same rule. Backup for P is Current for b, and vice-versa.

Comment (hugelgupf) [reply pkg/gpt/gpt.go]: Maybe it should be called something other than `Equal` then? `CorrespondingHeaders`? (too long) Idk. If you don't have better ideas, it looks good to merge.

**Comment by rminnich** [comment]
the names CurrentLBA and BackupLBA are from the standard, so changing them is not a good idea IMHO.

**Pull request merged**